components/CardList.tsx

```diff
@@ -0,0 +1,56 @@
+import { Box, Text, HStack, Flex, Wrap } from "@chakra-ui/react";
+import { ReactJSXElement } from "@emotion/react/types/jsx-namespace";
+import { PostWithUser } from "@lib/services/post";
+import { Book } from "@prisma/client";
+import BookCard from "./BookCard";
+import PostCard from "./PostCard";
+
+interface PostCardListProps {
+  posts: PostWithUser[] | undefined;
+}
+export function PostCardList({posts}: PostCardListProps) {
+  const items = posts?.map((post, i) => <PostCard key={i} {...post} />);
+  return <CardList items={items} itemName='Active Listing' />
+}
+
+interface BookCardListProps {
+  books: Book[] | undefined;
+}
+export function BookCardList({books}: BookCardListProps) {
+  const items = books?.map((book, i) => <BookCard key={i} {...book} />);
+  return <CardList items={items} itemName='Book' />
+}
```

Comment:
Consider just passing an empty array instead of making the inputs optional to make things a bit simpler